projects/packages/changelogger/src/Utils.php

```diff
@@ -188,9 +188,12 @@ public static function getRepoData( $file, OutputInterface $output, DebugFormatt
 				// PR number.
 				if ( isset( $cmd_output[1] ) ) {
 					$matches = array();
-					preg_match( '/\(#(\d+)\)$/', $cmd_output[1], $matches );
-					if ( isset( $matches[1] ) ) {
-						$repo_data['pr-num'] = $matches[1];
+					preg_match( '/(Merge pull request #(\d+))|(\(#(\d+)\))/', $cmd_output[1], $matches );
+					if ( isset( $matches[2] ) && ! empty( $matches[2] ) ) {
+						$repo_data['pr-num'] = $matches[2];
+					}
+					if ( isset( $matches[4] ) && ! empty( $matches[4] ) ) {
+						$repo_data['pr-num'] = $matches[4];
```

Comment:
We don't need so many capturing groups in the regex. OTOH, we should probably keep the anchoring.

Also, `empty()` already incorporates `isset()` logic, there's no need to use both here.
```suggestion
					preg_match( '/^Merge pull request #(\d+)|\(#(\d+)\)$/', $cmd_output[1], $matches );
					if ( ! empty( $matches[1] ) ) {
						$repo_data['pr-num'] = $matches[1];
					} elseif ( ! empty( $matches[2] ) ) {
						$repo_data['pr-num'] = $matches[2];
```

Reply:
Added in https://github.com/Automattic/jetpack/pull/27881/commits/4940548c2ca50db539244eb52fa3ddd784a814ce